### DISCUSSION TITLE
Error pairing esp32 to blueretro.io

### DISCUSSION THREAD

**Discussion by BaconExcitesMe** [darthcloud/BlueRetro#424] [Q&A]
I'm trying to pair my esp32 to blueretro.io but I keep getting an error. Also I cannot connect more then 1 controller.
![error](https://user-images.githubusercontent.com/104422703/194246184-b09ad1f4-e109-4f1b-a019-224ed26b97c9.JPG)

**Comment by vicsidious** [comment]
I got the same exact issue (reported it as issue [#431](https://github.com/darthcloud/BlueRetro/issues/431)] ) and had no answer yet, BUT, in case your problem is different than mine, make absolutely certain that every controller is disconnected from the ESP32 when you connect to the advance config page. 
There's a button on the the ESP32 that will drop all connections, do not only count on the fact that the core is in pairing mode for it to be able to connect. It could still be hooked to a controller.

From the wiki: 
![image](https://user-images.githubusercontent.com/115555660/195644606-57fc2048-4464-4e6f-8b28-354095557750.png)

Let me know if it solves your issue! 
-David

**Comment by darthcloud** [comment]
I made a fix for some of those connection issue:

Try this firmware:
[v1.8-rc3-1-gdd5d8bf_hw1.zip](https://github.com/darthcloud/BlueRetro/files/9966072/v1.8-rc3-1-gdd5d8bf_hw1.zip)

It may still fail once but so far It always succeed on 2nd try with that FW.

Also its always possible your RF environment might be too crowded, try to move the device around either closer or further away.
